### PR TITLE
Inherit the `prefixUrl` option from parent if it's `undefined`

### PR DESCRIPTION
**Pull request by Giotino**
#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [x] I have included some tests.

Fixes: #1444

### CODE REVIEW

**Comment by Giotino** [comment]
`prefixUrl: ''` is the default, but it was overwriting an explicit specified `prefixUrl` on `extend`.

Now (in this PR), `prefixUrl: ''` gets overwritten, but it's impossible to remove it from the options (I think this shouldn't be considered a problem).

The failing `cache` and `stream` tests are test that use `withServer` (which use `prefixUrl`) but not really need it, I'm going to remove it.
The last failing test is a `pagination` test, I'm working on it.

**Comment by Giotino** [comment]
The only breaking change (might not be so) that I could find is that with the pagination (if you're using `prefixUrl`) you can't change the URL returning something like this `{ url: 'http://something/?page=1234&token=abcd' }` in `pagination.paginate` unless you split the `url` into `prefixUrl` and `url` (in order to overwrite the `prefixUrl`):
```js
{ 
  prefixUrl: 'http://something/',
  url: '?page=1234&token=abcd'
}
```

File: source/core/index.ts [hunk 1]
```diff
@@ -1608,14 +1608,14 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 		options.password = options.password ?? '';
 
 		// `options.prefixUrl` & `options.url`
-		if (options.prefixUrl) {
+		if (options.prefixUrl && options.prefixUrl !== '') {
```
Comment (szmarczak) [inline source/core/index.ts:1611]: This is unnecessary as an empty string is `falsy` anyway.

**Comment by szmarczak** [comment]
> Now (in this PR), prefixUrl: '' gets overwritten, but it's impossible to remove it from the options (I think this shouldn't be considered a problem).

Actually, if we explicitly pass `prefixUrl: ''` then it should be empty. If it is `undefined` then it should take the previous value.

**Pull request merged**